src/resmom/linux/mom_start.c

```diff
@@ -2081,6 +2081,38 @@ set_job(job *pjob, struct startjob_rtn *sjr)
 					 */
 			sjr->sj_reservation = 0;
 		}
+		
+		/*
+		 * If we have a valid reservation id, save it off in case
+		 * the ALPS reservation information isn't part of the job struct
+		 * (e.g. job got killed before ALPS reservation ID was noted)
+		 * Then we can use this to cancel the ALPS reservation so we
+		 * don't leave orphaned ALPS reservations around.
+		 */
+		if(pjob->ji_extended.ji_ext.ji_reservation > 0) {
+			FILE	*fp = NULL;
+			char	filename[MAXPATHLEN];
+			int	rc;
+
+			if (rc = snprintf(filename, MAXPATHLEN, "%s/aux/%s.alpsresv", 
+				pbs_conf.pbs_home_path, pjob->ji_qs.ji_jobid) < MAXPATHLEN) {
+				fp = fopen(filename, "w");
+				if (fp == NULL) {
+					snprintf(log_buffer, sizeof(log_buffer),
+						"Unable to open %s", filename);
+					log_event(PBSEVENT_SYSTEM, PBS_EVENTCLASS_JOB, LOG_ERR, 
+						(char *)__func__, log_buffer);
+				} else {
+					fprintf(fp, "alps_resv_id %ld\n",pjob->ji_extended.ji_ext.ji_reservation);
```

Comment:
why do we need to store the string "alps_resv_id" in the file?

Reply:
The reservation ID will not be part of the job structure in case of a race condition for e.g. the job gets killed before reservation ID was noted. In that case we will need reservation ID to delete the reservation. So we are saving it in the file.

Reply:
either you have not understood the question, or you think I have not understood the solution. My question is why are we saving the **string** "alps_resv_id" in the file. As per your changes, we will be storing something of the sort "alps_resv_id 121537721" and I believe storing only "121537721" will suffice.